index.js

```diff
@@ -86,72 +84,59 @@ Collector.prototype._notifyIfFlushed = function(callback, err) {
     setImmediate(callback, err);
 };
 
-Collector.prototype._flushSeries = function(seriesName, points, callback) {
+Collector.prototype._flushPoints = function(points, callback) {
     if (!points || points.length === 0) {
         return;
     }
     var self = this;
 
     // only send N points at a time to avoid making requests too large
-    var spliceIndex;
-    if (self._client.protocol == 'udp:' || self._client.protocol == 'line') {
-      spliceIndex = self.computePointCountToSend(points.map(function (point) {
+    var spliceIndex = self.computePointCountToSend(points.map(function (point) {
         return JSON.stringify(point).length;
       }), MTU_SIZE);
-    } else {
-      spliceIndex = 50;
-    }
     var batch = points.slice(0, spliceIndex);
     points = points.slice(spliceIndex);
     var opt = { precision: self._time_precision };
 
     self._flushesInFlight++;
-    self._client.writePoints(seriesName, batch, opt, function(err) {
+    self._client.writePoints(batch, opt)
+      .then(function() {
         self._flushesInFlight--;
-        if (err) {
-            // TODO if error put points back to send again?
-            self.emit('error', err);
-            self._notifyIfFlushed(callback, err);
-            return;
-        }
-
         // there are more points to flush out
         if (points.length > 0) {
-            self._flushSeries(seriesName, points);
+            self._flushPoints(points);
         }
         self._notifyIfFlushed(callback);
-    });
+      }, function(err) {
+        self._flushesInFlight--;
+        // TODO if error put points back to send again?
+        self.emit('error', err);
+        self._notifyIfFlushed(callback, err);
+      });
 };
 
 Collector.prototype.flush = function(callback) {
-    var self = this;
-
-    Object.keys(self._series).forEach(function(key) {
-        var series = self._series[key];
-        delete self._series[key];
-        self._flushSeries(key, series, callback);
-    });
-    self._notifyIfFlushed(callback);
-};
-
-Collector.prototype._getSeries = function(name, reset) {
-    var series = this._series[name];
-    if (!series) {
-        series = [];
-        this._series[name] = series;
-    }
-    return series;
+    this._flushPoints(this._points, callback);
+    this._points = [];
 };
 
 // collect a data point (or object)
 // @param [Object] value the data
 // @param [Object] tags the tags (optional)
 Collector.prototype.collect = function(seriesName, value, tags) {
+    if (typeof value === 'number') {
```

Comment:
what about strings? Would it make sense to do `if (typeof value !== 'object')` instead?